dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java

```diff
@@ -260,12 +265,54 @@ private void setStatusCode(ContainerRequestContext requestContext, ContainerResp
 		}
 	}
 
+	private void handleError(ContainerResponseContext responseContext) {
+		if(isUnauthorizedError(responseContext)) {
+			setUnauthorizedErrorEntity(responseContext);
+		} else if(isThrowableError(responseContext)) {
+			setErrorEntity(responseContext);
+		}
+	}
+
+	private boolean isError(ContainerResponseContext responseContext) {
+		return isUnauthorizedError(responseContext) || isThrowableError(responseContext);
+	}
+
+	private boolean isThrowableError(ContainerResponseContext responseContext) {
+		return responseContext.getEntity()!=null && responseContext.getEntity() instanceof Throwable;
+	}
+
+	private boolean isUnauthorizedError(ContainerResponseContext responseContext) {
+		return responseContext.getEntity()!=null && responseContext.getEntity() instanceof String
+			&& responseContext.getStatus() == Response.Status.UNAUTHORIZED.getStatusCode();
+	}
+
 	private void setUnauthorizedErrorEntity(ContainerResponseContext responseContext) {
 		io.rtr.jsonapi.Error error = new io.rtr.jsonapi.Error();
-		error.setStatus(String.valueOf(Response.Status.UNAUTHORIZED.getStatusCode()));
-		error.setCode(String.valueOf(Response.Status.UNAUTHORIZED));
+		String numericalStatusCodeAsString = String.valueOf(Response.Status.UNAUTHORIZED.getStatusCode());
+		error.setStatus(numericalStatusCodeAsString);
+		error.setCode(numericalStatusCodeAsString);
 		error.setTitle(Response.Status.UNAUTHORIZED.getReasonPhrase());
 		error.setDetail(UNAUTHORIZED_ERROR_DETAILS);
 		responseContext.setEntity(new ErrorDocument(error), null, JSONAPI_MEDIATYPE);
 	}
+
+	private void setErrorEntity(ContainerResponseContext responseContext) {
+
+		Object entity = responseContext.getEntity();
+		assert(entity!=null && entity instanceof Throwable);
+
+		Throwable throwable = (Throwable) entity;
+		String statusCodeAsString = String.valueOf(responseContext.getStatus());
+		String reasonPhrase = responseContext.getStatusInfo() != null ?
+													responseContext.getStatusInfo().getReasonPhrase() : DEFAULT_STATUS.getReasonPhrase();
+		String applicationMessage = throwable.getCause() != null ?
+																throwable.getCause().getMessage() : throwable.getMessage();
+
+		Error error = new Error();
+		error.setStatus(statusCodeAsString); // Numerical status code, expressed as a String
+		error.setCode(statusCodeAsString); 		// Application-specific error code, expressed as a String
```

Comment:
Yes, I was kinda conflicted as I was refactoring this. And thinking about it again, I feel that what you mentioned makes sense.  Making the change now.